crates/ty_python_semantic/src/types/generics.rs

```diff
@@ -666,6 +681,9 @@ impl<'db> Specialization<'db> {
         for ty in self.types(db) {
             ty.find_legacy_typevars(db, typevars);
         }
+        if let Some(t) = self.tuple_inner(db) {
+            t.find_legacy_typevars(db, typevars);
+        }
```

Comment:
I honestly have no idea if this is a correct change or not. No tests fail if I remove this block, and I've struggled to find a behaviour difference from having this block or not having it.

This comment would suggest that it's incorrect to have this block here:

https://github.com/astral-sh/ruff/blob/ef1802b94f3bf7e7afcba2dfb9bd8896e73485c8/crates/ty_python_semantic/src/types/class.rs#L250-L258

But if I remove this, then all the `find_legacy_typevars` methods in `tuple.rs` are flagged as unused. Is it really okay to just delete them as part of this PR...?

Paging @dcreager for help!